bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleRegistryImpl.java

```diff
@@ -77,12 +81,22 @@ protected synchronized void removeProvider(Provider<Rule> provider) {
                 disabledRulesStorage.remove(uid);
             }
         }
+        if (provider instanceof ManagedRuleProvider) {
+            hasManagedRuleProvider = false;
+        }
         super.removeProvider(provider);
     }
 
     @Override
     public synchronized void add(Rule element) {
+        if (element == null) {
+            throw new IllegalArgumentException("The added rule must not be null!");
+        }
         String rUID = element.getUID();
+        if (!hasManagedRuleProvider) {
+            throw new IllegalStateException(
+                    "ManagedProvider is not available. The rule" + (rUID != null ? rUID : element) + "can't be added!");
```

Comment:
please add two spaces and apostrophes like:
`The rule '" + (rUID != null ? rUID : element) + "' can't be added!"`